packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx

```diff
@@ -23,38 +26,56 @@ function extractDomainFromEmail(email: string) {
 }
 
 export const CreateANewOrganizationForm = ({ slug }: { slug?: string }) => {
-  const { t, i18n } = useLocale();
+  const session = useSession();
+  if (!session.data) {
+    return null;
+  }
+  return <CreateANewOrganizationFormChild slug={slug} session={session} />;
+};
+
+const CreateANewOrganizationFormChild = ({
+  slug,
+  session,
+}: {
+  slug?: string;
+  session: Ensure<SessionContextValue, "data">;
+}) => {
+  const { t } = useLocale();
   const router = useRouter();
   const telemetry = useTelemetry();
   const [serverErrorMessage, setServerErrorMessage] = useState<string | null>(null);
-  const [showVerifyCode, setShowVerifyCode] = useState(false);
-
+  const isAdmin = session.data.user.role === UserPermissionRole.ADMIN;
+  const isImpersonated = session.data.user.impersonatedBy;
+  const defaultAdminEmail = session.data.user.email ?? "";
   const newOrganizationFormMethods = useForm<{
     name: string;
+    seats: number;
+    pricePerSeat: number;
     slug: string;
     adminEmail: string;
     adminUsername: string;
   }>({
     defaultValues: {
-      slug: `${slug ?? ""}`,
+      slug: deriveSlugFromEmail(defaultAdminEmail),
+      adminEmail: defaultAdminEmail,
+      name: deriveOrgNameFromEmail(defaultAdminEmail),
     },
   });
-  const watchAdminEmail = newOrganizationFormMethods.watch("adminEmail");
 
   const createOrganizationMutation = trpc.viewer.organizations.create.useMutation({
     onSuccess: async (data) => {
-      if (data.checked) {
-        setShowVerifyCode(true);
-      } else if (data.user) {
-        telemetry.event(telemetryEventTypes.org_created);
-        await signIn("credentials", {
-          redirect: false,
-          callbackUrl: "/",
-          email: data.user.email,
-          password: data.user.password,
+      telemetry.event(telemetryEventTypes.org_created);
+      // This is necessary so that server token has the updated upId
+      await session.update({
+        upId: data.upId,
+      });
+      if (isAdmin && data.userId !== session.data?.user.id) {
+        signIn("impersonation-auth", {
```

Comment:
Impersonate the user chosen as the organization owner, so that admin can now configure the organisation on his behalf. He won't need to have access to the org directly in this way.